### PR TITLE
Specify tracing span kind on creation

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
OpenTelemetry-OpenTracing bridge span kind can not be changed after creation, see https://github.com/open-telemetry/opentelemetry-go/issues/3953

The workaround is to specify span kind on creation which works for both Open Tracing and Open Telemetry bridge spans.

For #2104

### CODE REVIEW

File: filters/shedder/admission.go [hunk 1]
```diff
@@ -456,7 +456,6 @@ func (ac *admissionControl) startSpan(ctx context.Context) (span opentracing.Spa
 	if parent != nil {
 		span = ac.tracer.StartSpan(admissionControlSpanName, opentracing.ChildOf(parent.Context()))
 		ext.Component.Set(span, "skipper")
-		ext.SpanKind.Set(span, "shedder")
```
Comment (AlexanderYastrebov) [inline filters/shedder/admission.go:459]: I am not sure if we really need non-standard kind here

File: proxy/proxy.go [hunk 2]
```diff
@@ -958,14 +958,20 @@ func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Co
...
 	}
-	ctx.proxySpan = tracing.CreateSpan(spanName, req.Context(), p.tracing.tracer)
+
+	proxySpanOpts := []ot.StartSpanOption{ot.Tags{
+		SpanKindTag:       SpanKindClient,
+		SkipperRouteIDTag: ctx.route.Id,
+	}}
+	if parentSpan := ot.SpanFromContext(req.Context()); parentSpan != nil {
+		proxySpanOpts = append(proxySpanOpts, ot.ChildOf(parentSpan.Context()))
+
```
Comment (szuecs) [inline proxy/proxy.go:968]: whitespace

File: filters/shedder/admission.go (hunk 1 above)
Comment (szuecs) [reply filters/shedder/admission.go:459]: Can you check if we have streams that might select it and propose a change?

File: proxy/proxy.go (hunk 2 above)
Comment (AlexanderYastrebov) [reply proxy/proxy.go]: Fixed

File: filters/shedder/admission.go (hunk 1 above)
Comment (AlexanderYastrebov) [reply filters/shedder/admission.go:459]: Found one that also uses `operation IN ("admission_control")` which should be enough.

Comment (AlexanderYastrebov) [reply filters/shedder/admission.go:459]: Reached out to the owner and this was fixed.

**Comment by szuecs** [comment]
let's land it, no?

**Comment by RomanZavodskikh** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
Rebased and fixed span creation added by #3034

File: filters/shedder/admission.go (hunk 1 above)
Comment (szuecs) [reply filters/shedder/admission.go:459]: Can you add this to the release notes, too?
I don't know who will hit by that but that we do tell about.

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**